### ISSUE TITLE
How to compile under linux?

### ISSUE THREAD

**Issue by afsoeiro** [ahmadabdullah/BPA_EMTP#1]
Hi,

I currently want to compile this EMPT/ATP version using Gnu Fortran under Kubuntu Linux. I already have the necessary compiler software installed (gfortran package and needed dependencies), I believe.

I just want to know what kind of command I should issue to get it compiled. Looking through the code, I inferred the "COMPILE.COM" file is similar to makefiles we use in languages like C/C++. Is it correct? How do I get this code compiled?

Thanks in advance for all help.
André Soeiro

**Comment by ahmadabdullah** [comment]
I don’t think you will be able to compile it. The code has errors, unfortunately. The current git rep gives the status of EMTP more than 20 years ago and it is only good if you want to see the source code. ATP has progressed much beyond the EMTP but the main algorithm is the same. You can try to fix all errors and bing back if you are able to solve all issues so we can push back to git for future users.

Best Regards,
Ahmad Abdullah, PhD, PE

**Comment by afsoeiro** [comment]
I want to give it a try. The problem is I only have experience using 'make' or IDEs that do the compilation steps for me.

Do you know if there is a command I can issue to ask fortran to compile this code? Do you know how I run it under linux? In other words, how do I execute that COMPILE.COM file?

The COMPILE.LOG file does present lots of issues that probably come from the last compilation attempt. I want to be able to generate it out of a compilation execution in my machine.

Then, I will try to understand the basics of the code itself and solve those issues. 

My main objective is to join the mainstream EMTP/ATP development team, in order to get a nicer linux version running. I figure understanding this version first might help, as it may be smaller and less complex.

I am a currently registered EMTP/ATP user in Brazil.

Thanks again,
André Soeiro

**Comment by ahmadabdullah** [comment]
I haven’t tried it under Linux. The first step would be to compile under windows and see what error messages you get and go from there. However, there seem to be issues with array sizes used throughout.

Best Regards,
Ahmad Abdullah, PhD, PE

**Comment by Alan858** [comment]
Hi
I translated the codes to C++. You can find it here: github.com/Alan858/emtp-bpa-cpp.
Could you help to send a message to the ATP group  [ATP-EMTP-L:]? I did once, but seemed not go through.
Thanks,
Alan
